### PR TITLE
 [go] Bump images, dependencies and versions to go 1.22rc2

### PR DESCRIPTION
**Pull request by cpanato**
#### What type of PR is this?

/kind feature
/hold

#### What this PR does / why we need it:

- Bump images, dependencies and versions to go 1.22rc2

#### Which issue(s) this PR fixes:

Part of https://github.com/kubernetes/release/issues/3280

/assign @MadhavJivrajani @liggitt @puerco @xmudrii  
cc @kubernetes/release-managers 

#### Does this PR introduce a user-facing change?

```release-note
Kubernetes is now built with go 1.22rc2
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by MadhavJivrajani** [comment]
@thockin FYI ^

**Comment by liggitt** [comment]
change lgtm, other than commenting out the line in the dependencies yaml file so the minimum version line passes, is there anything left to merge this?

**Comment by thockin** [comment]
Thanks!  I'll rebase my beast as soon as this merges

**Comment by cpanato** [comment]
> change lgtm, other than commenting out the line in the dependencies yaml file so the minimum version line passes, is there anything left to merge this?

i will update it, thanks for the review

**Comment by cpanato** [comment]
ready @liggitt 
thanks!

File: staging/publishing/rules.yaml [hunk 1]
```diff
@@ -2392,4 +2392,4 @@ rules:
       - staging/src/k8s.io/endpointslice
 recursive-delete-patterns:
 - '*/.gitattributes'
-default-go-version: 1.21.6
+default-go-version: 1.22rc2
```
Comment (liggitt) [inline staging/publishing/rules.yaml:2395]: does this change the go version we publish to modules for patch tags? If we, we want to keep this at 1.21.x, not an rc

**Comment by thockin** [comment]
Now that 1.22.0 is out, is someone bumping to the .0 ?

**Comment by dims** [comment]
@thockin https://github.com/kubernetes/release/issues/3280

**Pull request merged**